Result/ResultType.swift

```diff
@@ -101,3 +101,29 @@ infix operator &&& {
 public func &&& <L: ResultType, R: ResultType where L.Error == R.Error> (left: L, @autoclosure right: () -> R) -> Result<(L.Value, R.Value), L.Error> {
 	return left.flatMap { left in right().map { right in (left, right) } }
 }
+
+/// Return a array of `V` by applying transform to all elements of array 'U' and filtering out failure ones
+public func mapFilter<U, V, Error>(results: [U], @noescape transform: U -> Result<V, Error>) -> [V]{
```

Comment:
Great response, @robrix !
And I agree that a particular `mapFilter` is not necessary here.

BTW: my first version of `mapFilter' is something like 'values.lazy.map(transform).filter(...).map{$0!}' which is far more complicated than the version your provided.
:+1